lingua_franca/lang/parse_en.py

```diff
@@ -609,30 +618,229 @@ def extract_duration_en(text):
+                value = relative_units[unit]
+                _leftover, _ = math.modf(value)
+                if _leftover != 0:
+                    raise ValueError("relativedelta requires {unit} to be an "
+                                     "integer".format(unit=unit))
+                relative_units[unit] = int(value)
+        duration = relativedelta(**relative_units) if \
+            any(relative_units.values()) else None
+    else:
+        microseconds = 0
+        units = ['months', 'years', 'decades', 'centurys', 'millenniums',
+                 "microseconds", "milliseconds", "seconds", "minutes",
+                 "hours", "days", "weeks"]
+
+        for unit in units:
+            unit_pattern = pattern.format(
+                unit=unit[:-1])  # remove 's' from unit
+            matches = re.findall(unit_pattern, text)
+            value = sum(map(float, matches))
+            text = re.sub(unit_pattern, _replace_token, text)
+            if unit == "microseconds":
+                microseconds += value
+            elif unit == "milliseconds":
+                microseconds += value * 1000
+            elif unit == "seconds":
+                microseconds += value * 1000 * 1000
+            elif unit == "minutes":
+                microseconds += value * 1000 * 1000 * 60
+            elif unit == "hours":
+                microseconds += value * 1000 * 1000 * 60 * 60
+            elif unit == "days":
+                microseconds += value * 1000 * 1000 * 60 * 60 * 24
+            elif unit == "weeks":
+                microseconds += value * 1000 * 1000 * 60 * 60 * 24 * 7
+            elif unit == "months":
+                microseconds += value * 1000 * 1000 * 60 * 60 * 24 * \
+                                DAYS_IN_1_MONTH
+            elif unit == "years":
+                microseconds += value * 1000 * 1000 * 60 * 60 * 24 * \
+                                DAYS_IN_1_YEAR
+            elif unit == "decades":
+                microseconds += value * 1000 * 1000 * 60 * 60 * 24 * \
+                                DAYS_IN_1_YEAR * 10
+            elif unit == "centurys":
+                microseconds += value * 1000 * 1000 * 60 * 60 * 24 * \
+                                DAYS_IN_1_YEAR * 100
+            elif unit == "millenniums":
+                microseconds += value * 1000 * 1000 * 60 * 60 * 24 * \
+                                DAYS_IN_1_YEAR * 1000
+
+        if resolution == DurationResolution.TOTAL_MICROSECONDS:
+            duration = microseconds
+        elif resolution == DurationResolution.TOTAL_MILLISECONDS:
+            duration = microseconds / 1000
+        elif resolution == DurationResolution.TOTAL_SECONDS:
+            duration = microseconds / (1000 * 1000)
+        elif resolution == DurationResolution.TOTAL_MINUTES:
+            duration = microseconds / (1000 * 1000 * 60)
+        elif resolution == DurationResolution.TOTAL_HOURS:
+            duration = microseconds / (1000 * 1000 * 60 * 60)
```

Comment:
Using the `unit_to_multiplier` map suggested above would save defining these twice. You could just use `duration = microseconds / unit_to_multiplier[unit]` here.